components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -574,14 +575,19 @@ const RepositoryInputError: FC<RepositoryInputErrorProps> = ({ title, message, l
 export const RepositoryNotFound: FC<{ error: StartWorkspaceError }> = ({ error }) => {
     const { host, owner, userIsOwner, userScopes = [], lastUpdate } = error.data || {};
 
-    const authProviders = useAuthProviders();
+    const authProviders = useAuthProviderDescriptions();
     const authProvider = authProviders.data?.find((a) => a.host === host);
     if (!authProvider) {
         return <RepositoryInputError title="The repository was not found in your account." />;
     }
 
     // TODO: this should be aware of already granted permissions
-    const missingScope = authProvider.authProviderType === "GitHub" ? "repo" : "read_repository";
+    const missingScope =
+        authProvider.type === AuthProviderType.GITHUB
+            ? "repo"
+            : authProvider.type === AuthProviderType.GITLAB
+            ? "api"
+            : "";
```

Comment:
Nice catch! 
I tried to fix the previous state, but didn't finish it apparently. 

Old logic reads like this: If type is GitHub use GH scope, otherwise use GitLab scope. This is obviously wrong for BB and BBS. 

The update reads like: this: if type is GitHub use GH scope, otherwise if type is GitLab use GL scope, otherwise use whatever is default for the provider type, which is known on server side. 

I think that state is already a small fix. It would be great to verify though.

Reply:
> I think that state is already a small fix. It would be great to verify though.

Do we have any account (like in 1Password) that could test those providers?

Reply:
BBS account in 1P, otherwise we can patch the deployment's secret in K8s to remove GitLab/Bitbucket from the default list. This gives us the change to create them in-App.